lib/internal/http2/compat.js

```diff
@@ -221,7 +222,7 @@ class Http2ServerRequest extends Readable {
   _read(nread) {
     const stream = this[kStream];
     if (stream !== undefined) {
-      stream.resume();
+      process.nextTick(() => stream.resume());
```

Comment:
There's a race condition where `pause` can get evaluated in between `resume` being called and actually running (basically after `resumeScheduled: true` and before nextTick). This fixes it and passes all our tests.

Reply:
we should not create a closure here, but rather use a top level function and pass the stream in there.